### PR TITLE
Add support for adc on STM32 G0 Series

### PR DESCRIPTION
**Pull request by str4t0m**
- Add support for ADC on G0 series.
- Enables ADC on Nucleo G071RB board (ADC1 channel 0 and 1).

### CODE REVIEW

File: dts/arm/st/g0/stm32g0.dtsi [hunk 1]
```diff
@@ -198,6 +198,28 @@
...
+			#size-cells = <0>;
+			reg = <0x40013000 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00001000>;
+			interrupts = <25 0>;
+			status = "disabled";
+			label = "SPI_1";
+		};
+
+		spi2: spi@40003800 {
+			compatible = "st,stm32-spi";
```
Comment (erwango) [inline dts/arm/st/g0/stm32g0.dtsi:214]:
```suggestion
			compatible = "st,stm32-spi-fifo", "st,stm32-spi";
```

**Review by erwango — CHANGES_REQUESTED** [review]
@str4t0m Thanks for this contribution!
Aside from a comment, the 4 first commits are ok for me, but need more time for the ADC related stuff. I propose to make a dedicated PR for these (DAC, SPI), so they can be merged faster. And convert this PR into an official PR.

**Comment by str4t0m** [comment]
Thanks, I have fixed the spi compatible and picked the spi and dac commits in the mentioned new pr.
For convenience I have left these commits also in this pr, please let me know if I should better remove them here.

File: dts/arm/st/g0/stm32g0.dtsi (hunk 1 above)
Comment (str4t0m) [reply dts/arm/st/g0/stm32g0.dtsi]: I have added this with the latest commits.

**Comment by erwango** [comment]
> Thanks, I have fixed the spi compatible and picked the spi and dac commits in the mentioned new pr.
> For convenience I have left these commits also in this pr, please let me know if I should better remove them here.

@str4t0m Let's remove the spi/dac commits from here, if not too much trouble

**Review by FRASTM — APPROVED** [review]

**Comment by erwango** [comment]
@str4t0m please rebase

**Comment by str4t0m** [comment]
rebased

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
DNM as CI reports issues

**Comment by erwango** [comment]
@str4t0m can you check issues reported by CI ?

**Comment by str4t0m** [comment]
Sorry, one brace got lost during rebase this is now fixed.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**